cirq-core/cirq/transformers/merge_single_qubit_gates.py

```diff
@@ -152,3 +159,184 @@ def merge_func(m1: 'cirq.Moment', m2: 'cirq.Moment') -> Optional['cirq.Moment']:
         deep=context.deep if context else False,
         tags_to_ignore=tuple(tags_to_ignore),
     ).unfreeze(copy=False)
+
+
+def _values_of_sweep(sweep: Sweep, key: str | sympy.Symbol):
+    p = sympy.Symbol(key) if isinstance(key, str) else key
+    return [resolver.value_of(p) for resolver in sweep]
+
+
+@transformer_api.transformer
+def merge_single_qubit_gates_to_phxz_symbolized(
+    circuit: 'cirq.AbstractCircuit',
+    *,
+    context: Optional['cirq.TransformerContext'] = None,
+    sweep: Sweep,
+    atol: float = 1e-8,
+) -> Tuple['cirq.Circuit', Sweep]:
+    """Merge consecutive single qubit gates as PhasedXZ Gates. Symbolized if any of the consecutive gates is symbolized.
+
+    Example:
+        # pylint: disable=line-too-long
+        >>> q0, q1 = cirq.LineQubit.range(2)
+        >>> c = cirq.Circuit(cirq.X(q0),cirq.CZ(q0,q1)**sympy.Symbol("cz_exp"),cirq.Y(q0)**sympy.Symbol("y_exp"),cirq.X(q0))
+        >>> print(c)
+        0: ───X───@──────────Y^y_exp───X───
+                  │
+        1: ───────@^cz_exp─────────────────
+        >>> new_circuit, new_sweep = cirq.merge_single_qubit_gates_to_phxz_symbolized(\
+                c, sweep=cirq.Points(key="cz_exp", points=[0, 1]) * cirq.Points(key="y_exp", points=[0, 1])\
+            )
+        >>> print(new_circuit)
+        0: ───PhXZ(a=-1,x=1,z=0)───@──────────PhXZ(a=a0,x=x0,z=z0)───
+                                   │
+        1: ────────────────────────@^cz_exp──────────────────────────
+        >>> print(new_sweep)
+        cirq.Points('z0', [0, -1.0, 0, -1.0]) + cirq.Points('x0', [1, 0.0, 1, 0.0]) + cirq.Points('a0', [-1.0, -0.5, -1.0, -0.5]) + cirq.Points('cz_exp', [0, 0, 1, 1])
+        # pylint: disable=line-too-long
```

Comment:
@eliottrosenberg , I think this version should work as expected, I have tested the code in the unit tests and here is one example, before I move on to fix the PR, could you help test and check if the function returns as expected?